stanford_soe_r25.pages.inc

```diff
@@ -0,0 +1,994 @@
+<?php
+
+// the menu page callback function to build the main content for a calendar page
+function _stanford_soe_r25_calendar_page($room = NULL) {
+
+  // get default view and date params from URL if provided by a permalink
+  $params = array();
+  foreach (drupal_get_query_parameters() as $key => $param) {
+    $params[$key] = check_plain($param);
+  }
+
+  // load our room configuration info and prepare an $output string
+  $rooms = _stanford_soe_r25_room_config_load();
+  $output = array();
+
+  // only output a page if we have the requested room, otherwise output MENU_NOT_FOUND
+  if (!empty($room) && isset($rooms[$room]['status'])) {
+    // room status is disabled, view-only, tentative reservable or confirmed reservable
+    $status = intval($rooms[$room]['status']);
+
+    // make sure we have an enabled room, otherwise display an error msg (below)
+    if ($status > stanford_soe_r25_ROOM_STATUS_DISABLED) {
+      // set some JavaScript variables to be used at the browser by stanford_soe_r25_fullcall.js
+      drupal_add_js(array('stanfordR25Room' => $room), 'setting');
```

Comment:
why not combine all these settings into a single array and add the js settings at the end of the function